File: how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        idx_counts = {}
        for i in range(len(nums)):
            idx_counts[i] = 0
            for j in range(len(nums)):
                if nums[j] < nums[i]:
                    idx_counts[i] += 1
        counts = []
        for i in range(len(nums)):
            counts.append(idx_counts[i])
        return counts
        

File: how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        unordered_map <int, int> idx_counts;
        for(int i = 0; i < nums.size(); i++){
            idx_counts[i] = 0;
            for(int j = 0; j < nums.size(); j++)
                if(nums[j] < nums[i])
                    idx_counts[i]++;
        }
        vector <int> counts;
        for(int i = 0; i < nums.size(); i++)
            counts.push_back(idx_counts[i]);
        return counts;
    }
};
